File: src/main/java/io/zeitwert/fm/dms/model/enums/CodeDocumentCategory.java
package io.zeitwert.fm.dms.model.enums;

import io.zeitwert.ddd.enums.model.base.EnumeratedBase;
import lombok.Data;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder
public final class CodeDocumentCategory extends EnumeratedBase {

	@Override
	public boolean equals(Object other) {
		return super.equals(other);
	}

	@Override
	public int hashCode() {
		return super.hashCode();
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/dms/model/enums/CodeDocumentCategory.kt
package io.zeitwert.fm.dms.model.enums

import dddrive.ddd.model.EnumeratedEnum
import dddrive.ddd.model.base.EnumerationBase

enum class CodeDocumentCategory(
	override val defaultName: String,
) : EnumeratedEnum {

	AVATAR("Avatar"),
	BANNER("Banner"),
	FOTO("Foto"),
	LOGO("Logo"),
	;

	override val enumeration get() = Enumeration

	companion object Enumeration :
		EnumerationBase<CodeDocumentCategory>(CodeDocumentCategory::class.java) {

		init {
			entries.forEach { addItem(it) }
		}

		@JvmStatic
		fun getDocumentCategory(itemId: String?) = if (itemId != null) getItem(itemId) else null
	}
}
